### DISCUSSION TITLE
customization of magic link with docker host

### DISCUSSION THREAD

**Discussion by Mats-Hjalmar** [supabase/supabase#17961] [Questions]
I can't find the setting to configure to change the domain of where my magic link is being sent to. Right now it looks like this

http://kong/auth/v1/verify?token=[SOME-TOKEN]&type=magiclink&redirect_to=com.example.supabase_login_app://signin-callback/

if i replace kong for localhost:8000 (my studio url) i get Kong Error No API key found in request.

How can i configure it?

**Comment by s0h311** [comment]
Hi @Mats-Hjalmar navigate to Authentication > URL Configuration and make sure you have the correct URL.

**Comment by Mats-Hjalmar** [comment]
Im using supabase on docker and do not have that config option. I only have
"policies"

**Comment by JLAcostaEC** [comment]
I have the same problem. I have not been able to change the email route. Always send http://kong/auth/v1/verify/.....

I have configured the following in the ENV File:

MAILER_REDIRECT_URI="http://127.0.0.1:3000/"
MAILER_URLPATHS_INVITE="http://127.0.0.1:8000/"
MAILER_URLPATHS_CONFIRMATION="http://127.0.0.1:8000/"
MAILER_URLPATHS_RECOVERY="http://127.0.0.1:8000/"
MAILER_URLPATHS_EMAIL_CHANGE="http://127.0.0.1:8000/"
SITE_URL="http://127.0.0.1:3000/"

But I can't get the sending of these emails from Supabase Studio to have the correct URL in the Email. I've searched for a while with no good results. Is there any way to correct this? Or activate the editing of email templates from Supabase Studio?

**Comment by lichong-a** [comment]
same problem

**Reply by JLAcostaEC** [reply]
I was able to solve this, however, less than a week ago I formatted my PC and lost the configuration, I will retake this topic and leave you the answer to this case since at this moment I do not remember it

**Comment by JLAcostaEC** [comment]
At this time, I am unable to recreate the solution I had created. But here you have 2 options in: #5971

1. [Add the base URL at `docker-compse.yml` file](https://github.com/supabase/supabase/issues/5971#issuecomment-1073602195)

> I got the solution
> 
> add the base url of __MAILER_URLPATHS__ params in docker-compse.yml:
> 
> GOTRUE_MAILER_URLPATHS_INVITE: **http://localhost:8000**/auth/v1/verify GOTRUE_MAILER_URLPATHS_CONFIRMATION: **http://localhost:8000**/auth/v1/verify GOTRUE_MAILER_URLPATHS_RECOVERY: **http://localhost:8000**/auth/v1/verify GOTRUE_MAILER_URLPATHS_EMAIL_CHANGE: **http://localhost:8000**/auth/v1/verify

Personally, I prefer to use `GOTRUE_MAILER_URLPATHS_INVITE: ${API_EXTERNAL_URL}${MAILER_URLPATHS_INVITE}`, so you can define both with the same result. (Obviously adding the rest of the cases)


2. [Add the base URL at `.env` file](https://github.com/supabase/supabase/issues/5971#issuecomment-1074879157)

> @jesper-bylund you can add param API_EXTERNAL_URL in .env API_EXTERNAL_URL=http://localhost:8000
> 
> then update docker-compose.yml GOTRUE_MAILER_URLPATHS_INVITE: ${API_EXTERNAL_URL}/auth/v1/verify GOTRUE_MAILER_URLPATHS_CONFIRMATION: ${API_EXTERNAL_URL}/auth/v1/verify GOTRUE_MAILER_URLPATHS_RECOVERY: ${API_EXTERNAL_URL}/auth/v1/verify GOTRUE_MAILER_URLPATHS_EMAIL_CHANGE: ${API_EXTERNAL_URL}/auth/v1/verify
> 
> same solution here: [#5970 (comment)](https://github.com/orgs/supabase/discussions/5970#discussion-3951198)

**Reply by lichong-a** [reply]
This indeed solved the problem! Thank you so much!
